### PR TITLE
Gem modernisation and Rails 6 updates

### PR DESCRIPTION
**Pull request by alexdunae**
A bunch of minor gem modernisation stuff in here, as well as changes for `distinct` vs `uniq` and association cache clearing.

### CODE REVIEW

File: lib/acts_as_taggable/acts_as_taggable.rb [hunk 1]
```diff
@@ -114,24 +114,27 @@ def tag_names
...
     end
 
-    def reset_scoped_associations(record)
-      return unless record.tag_type
-      send("#{record.tag_type}_taggings").reset
-      send("#{record.tag_type}_tags").reset
-    end
-
-    def reset_associations(record)
+    def reset_assocations_after_tag_change(tag)
```
Comment (rywall) [inline lib/acts_as_taggable/acts_as_taggable.rb:128]:
```suggestion
    def reset_associations_after_tag_change(tag)
```

File: lib/acts_as_taggable/acts_as_taggable.rb [hunk 2]
```diff
@@ -114,7 +114,8 @@ def tag_names
     end
 
     def tag_names=(names)
-      send HelperMethods.scoped_association_assignment_name, names.select(&:present?).collect {|tag_name| self.class.create_tag(tag_name) }
+      tag_objects = names.select(&:present?).collect {|tag_name| self.class.create_tag(tag_name) }.uniq
```
Comment (rywall) [inline lib/acts_as_taggable/acts_as_taggable.rb:117]: Why are we not calling `uniq` on the `names` before we call `collect { create_tag }`?

File: lib/acts_as_taggable/acts_as_taggable.rb [hunk 3]
```diff
@@ -114,24 +114,27 @@ def tag_names
...
-    def reset_associations(record)
+    def reset_assocations_after_tag_change(tag)
       send(:taggings).reset
       send(:tags).reset
+      return unless tag.tag_type
+      send("#{tag.tag_type}_taggings").reset
+      send("#{tag.tag_type}_tags").reset
+    end
+
+    def reset_assocations_after_taggings_change(tagging)
```
Comment (rywall) [inline lib/acts_as_taggable/acts_as_taggable.rb:136]:
```suggestion
    def reset_associations_after_taggings_change(tagging)
```

**Review by rywall — CHANGES_REQUESTED** [review]
Looks good. :) A couple small fixes/questions.

Next time we talk let's quickly discuss aea51068590a31e65a13424ad2fda170c54bcb88. I'm having trouble figuring out the reason behind the fix.

File: lib/acts_as_taggable/acts_as_taggable.rb (hunk 2 above)
Comment (alexdunae) [reply lib/acts_as_taggable/acts_as_taggable.rb:117]: That could make sense, too.  Is there a possible case where `Tag#sanitize` does some of its magic and gives us duplicates, e.g. `names = ["Tag ", " tag ", "   tAg"]` so we'd want to call `uniq` on the actual tag records?

File: lib/acts_as_taggable/acts_as_taggable.rb (hunk 1 above)
Comment (alexdunae) [reply lib/acts_as_taggable/acts_as_taggable.rb]: Should be good now

File: lib/acts_as_taggable/acts_as_taggable.rb (hunk 2 above)
Comment (rywall) [reply lib/acts_as_taggable/acts_as_taggable.rb:117]: Ah yes good point. This is better than my suggestion then :).

**Pull request merged**